core/state/state_object.go

```diff
@@ -311,28 +335,49 @@ func (s *StateObject) GetCommittedState(db Database, key common.Hash) common.Has
 		value.SetBytes(content)
 	}
 	s.setOriginStorage(key, value)
-	return value
+	return value, nil
 }
 
 // SetState updates a value in account storage.
-func (s *StateObject) SetState(db Database, key, value common.Hash) {
+func (s *StateObject) SetState(db Database, key, value common.Hash) error {
 	// If the fake storage is set, put the temporary state update here.
 	if s.fakeStorage != nil {
 		s.fakeStorage[key] = value
-		return
+		return nil
 	}
 	// If the new value is the same as old, don't set
-	prev := s.GetState(db, key)
+	prev, err := s.GetState(db, key)
+	if exErr, ok := err.(*ExpiredStateError); ok {
+		exErr.isInsert = true
```

Comment:
from my understanding,`isInsert` is used to trace if the expired error is returned when trying to update state. Will there be any other use cases, other than a different error string?

Reply:
No, just using it to output a reasonable err msg.